barcodes_stock_inventory/models/models.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models
+
+
+class BarcodeStockInventory(models.Model):
+
+    _name = 'stock.inventory'
+    _inherit = ['stock.inventory', 'barcodes.barcode_events_mixin']
+
+    def on_barcode_scanned(self, barcode):
+
+        location = self.env['stock.location'].search([('barcode', '=', barcode)])
+        if location:
+            self.location_id = location[0]
+            return
+
+        product = self.env['product.product'].search([('barcode', '=', barcode)])[0]
+        if product:
+            inventory = self.search([('state', '=', 'confirm'), ('location_id', '=', self.location_id.id)])
+            line = False
+            lines = self.line_ids.search([('state', '=', 'confirm'),
```

Comment:
``self.line_ids.search`` is the same as  ``self.env.['stock.inventory.line'].search``. Use later form, because first one may confuse